pkg/k8s/defaulter.go

```diff
@@ -29,35 +34,59 @@ type Defaulter interface {
 // StateIntoSpecDefaulter contains the required 'defaultValue' field and the
 // optional 'userOverride' field.
 type StateIntoSpecDefaulter struct {
-	defaultValue string
-	userOverride *string
+	client client.Client
 }
 
-func NewStateIntoSpecDefaulter(defaultValue string, userOverride *string) (Defaulter, error) {
-	if !isAcceptedValue(defaultValue, StateIntoSpecAnnotationValues) {
-		return nil, fmt.Errorf("invalid default value '%v' for '%v' annotation, need to be one of {%v}", defaultValue, StateIntoSpecAnnotation, strings.Join(StateIntoSpecAnnotationValues, ", "))
-	}
-	if userOverride != nil && !isAcceptedValue(*userOverride, StateIntoSpecAnnotationValues) {
-		return nil, fmt.Errorf("invalid user override value '%v' for '%v' annotation, need to be one of {%v}", userOverride, StateIntoSpecAnnotation, strings.Join(StateIntoSpecAnnotationValues, ", "))
-	}
+func NewStateIntoSpecDefaulter(client client.Client) Defaulter {
 	return &StateIntoSpecDefaulter{
-		defaultValue: defaultValue,
-		userOverride: userOverride,
-	}, nil
+		client: client,
+	}
 }
 
-func (v *StateIntoSpecDefaulter) ApplyDefaults(_ context.Context, resource client.Object) (changed bool, err error) {
+func (v *StateIntoSpecDefaulter) ApplyDefaults(ctx context.Context, resource client.Object) (changed bool, err error) {
+	cccNamespacedName := types.NamespacedName{
+		Namespace: resource.GetNamespace(),
+		Name:      operatork8s.ConfigConnectorContextAllowedName,
+	}
+	ccc, err := v.getConfigConnectorContext(ctx, cccNamespacedName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("error getting ConfigConnectorContext object %v/%v: %w", cccNamespacedName.Namespace, cccNamespacedName.Name, err)
+	}
+	if ccc.APIVersion != "v1beta1" {
```

Comment:
Harmless, but you don't need to check this, you requested v1beta1 so you'll always get v1beta1

Reply:
Good point! Removed.